mem0/memory/main.py

```diff
@@ -1429,7 +1435,7 @@ async def _create_memory(self, data, existing_embeddings, metadata=None):
         capture_event("mem0._create_memory", self, {"memory_id": memory_id, "sync_type": "async"})
         return memory_id
 
-    async def _create_procedural_memory(self, messages, metadata=None, llm=None, prompt=None):
+    async def _create_procedural_memory(self, messages, metadata: dict | None = None, llm=None, prompt=None):
```

Comment:
Change this to `Optional[dict]` because this project supports Python >=3.9 and `|` was introduced in Python 3.10